docs/api-docs.rst

```diff
@@ -1,6 +1,217 @@
+        (, "input": <input id to bind to for a step>)
+        (, "input_step": <input step name to connect>)
+        (, "output_step": <output step name to connect>)
+    }
+
+The workflow mode simply allows for a directed acyclic graph of tasks to be
+specified to :py:func:`romanesco.run`.
+
+.. code-block:: none
+
+    <TASK_INPUT> ::= {
+        "id": <string, the variable name>,
+        "type": <data type>,
+        "format": <data format>
+        (, "target": <INPUT_TARGET_TYPE>)   ; default is "memory"
+        (, "filename": <name of file if target="filepath">)
+    }
+
+    <INPUT_TARGET_TYPE> ::= "memory" | "filepath"
+
+    <TASK_OUTPUT> ::= {
+        "id": <string, the variable name>,
+        "type": <data type>,
+        "format": <data format>
+        (, "target": <INPUT_TARGET_TYPE>)   ; default is "memory"
+    }
+
+
+The input specification
+***********************
+
+The ``inputs`` argument to :py:func:`romanesco.run` specifies the inputs to the
+task described by the ``task`` argument. Specifically, it tells what data should
+be placed into the task input ports.
+
+.. code-block:: none
+
+    <INPUTS> ::= {
+        <id> : <INPUT_BINDING>
+    }
+
+The input spec is a dictionary mapping each ``id`` (corresponding to the ``id`` key of
+each task input) to its data binding for this execution.
+
+.. code-block:: none
+
+    <INPUT_BINDING> ::= <INPUT_BINDING_HTTP> | <INPUT_BINDING_LOCAL> |
+                        <INPUT_BINDING_MONGODB> | <INPUT_BINDING_INLINE>
+
+    <INPUT_BINDING_HTTP> ::= {
+        "mode": "http",
+        "url": <url of data to download>
+        (, "headers": <dict of HTTP headers to send when fetching>)
+        (, "method": <http method to use, default is "GET">)
+        (, "maxSize": <integer, max size of download in bytes>)
+    }
+
+The http input mode specifies that the data should be fetched over HTTP. Depending
+on the ``target`` field of the corresponding task input specifier, the data will
+either be passed in memory, or streamed to a file on the local filesystem, and the
+variable will be set to the path of that file.
+
+.. code-block:: none
+
+    <INPUT_BINDING_LOCAL> ::= {
+        "mode": "local",
+        "path": <path on local filesystem to the file>
+    }
+
+The local input mode denotes that the data exists on the local filesystem. Its
+contents will be read into memory and the variable will point to those contents.
+
+.. code-block:: none
+
+    <INPUT_BINDING_MONGODB> ::= {
+        "mode": "mongodb",
+        "db": <the database to use>,
+        "collection": <the collection to fetch from>
+        (, "host": <mongodb host, default is "localhost">)
+    }
+
+The mongodb input mode specifies that the data should be fetched from a mongo
+collection. This simply binds the entire BSON-encoded collection to the input
+variable.
+
+.. code-block:: none
+
+    <INPUT_BINDING_INLINE> ::= {
+        "mode": "inline",
+        "data": <data to bind to the variable>
+    }
```

Comment:
See this bit of code:

https://github.com/Kitware/romanesco/blob/master/romanesco/__init__.py#L192